draft-barnes-acme.md

```diff
@@ -141,10 +159,29 @@ Some HTTPS bodies in ACME are authenticated and integrity-protected by being enc
 * If the JWS is in the JSON Serialization, it MUST NOT include more than one signature in the "signatures" array
 * The JWS Header MUST include "alg" and "jwk" fields
 
-
-# Protocol Overview
-
-ACME allows a client to request certificate management actions using a set of JSON messages carried over HTTPS.   In some ways, ACME functions much like a traditional CA, in which a user creates an account, adds identifiers to that account (proving control of the domains), and requests certificate issuance for those domains while logged in to the account.  
+In order to prevent an attacker from replaying ACME messages against
+different ACME resources, or at a different time than they were sent,
+every authenticated ACME request body MUST include, in the Protected
+header, two Private Header Parameters {{RFC7515}}: "acmeURL" and
+"acmeNonce". The value of "acmeURL" MUST be equal to the absolute
+URL to which the client is submitting the request. The value of "acmeNonce" MUST be
+equal to a nonce previously received from the server in the ACME-Nonce
+HTTP header of any message. The client MUST NOT reuse nonces. If a
+client needs to send a message but does not have a fresh nonce, it
+may request any GET resource from the server in order to receive one.
+
+The ACME server MUST send an ACME-Nonce HTTP header in response to any client
+request, including GET requests. The header MUST be unique in each response,
```

Comment:
This seems like an odd design and likely to result in a lot of stale nonces. Why not have an explicit way to request nonces.